otter/convergence/model.py

```diff
@@ -126,6 +154,69 @@ def __init__(self):
         assert self.state in ServerState.iterconstants(), \
             "%r is not a ServerState" % (self.state,)
 
+    @classmethod
+    def group_id_from_metadata(cls, metadata):
```

Comment:
Sure - if we implement https://github.com/rackerlabs/otter/issues/1106 though, should it be an instance function?  We can lazy-parse (parse an attribute only when it's needed)